src/modules/wasi/WspOneModule.v3

```diff
@@ -174,159 +197,148 @@ class WspOneModule extends HostModule("wasi_snapshot_preview1") {
-			if (iobuf < 0) return i(WspOneErrno.INVAL.code); // Invalid memory
+			var iobuf = getRegionPtr(int.!(iov_ptr), int.!(iov_len));
+			if (iobuf < 0) return INVAL_HR; // Invalid memory
 			var out = memory.writeOut(wfd.sysfd, u32.view(iobuf), iov_len);
 			if (out < 0) return i(WspOneErrno.IO.code);
 			total += out;
 			iovbuf += 8;
 		}
 		memory.write_u32(u32.view(outsize_ptr), 0, u32.view(total));
-		return i(WspOneErrno.SUCCESS.code);
+		return SUCCESS_HR;
 	}
-	def path_create_directory(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def path_filestat_get(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def path_filestat_set_times(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def path_link(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def path_open(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def path_readlink(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def path_remove_directory(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def path_rename(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def path_symlink(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def path_unlink_file(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def clock_res_get(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def clock_time_get(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def sched_yield(args: Range<Value>) -> HostResult { return UNIMPLEMENTED; }
-	def proc_exit(args: Range<Value>) -> HostResult {
-		return HostResult.Throw(Exit.new(Values.unbox_u(args[0])));
+	def path_create_directory(fd: int, path: int, path_len: int) -> int;
+	def path_filestat_get(fd: int, flags: int, path: int, path_len: int, buf: int) -> int;
+	def path_filestat_set_times(fd: int, flags: int, path: int, path_len: int, st_atim: long, st_mtim: long, fst_flags: int) -> int;
+	def path_link(old_fd: int, old_flags: int, old_path: int, old_path_len: int, new_fd: int, new_path: int, new_path_len: int) -> int;
+	def path_open(dirfd: int, dirflags: int, path: int, path_len: int, o_flags: int, fs_rights_base: long, fs_rights_inheriting: long, fs_flags: int, fd: int) -> int;
+	def path_readlink(dir_fd: int, path: int, path_len: int, buf: int, buf_len: int, buf_used: int) -> HostResult;
+	def path_remove_directory(fd: int, path: int, path_len: int) -> HostResult;
+	def path_rename(old_fd: int, old_path: int, old_path_len: int, new_fd: int, new_path: int, new_path_len: int) -> int;
+	def path_symlink(old_path: int, old_path_len: int, fd: int, new_path: int, new_path_len: int) -> int;
+	def path_unlink_file(fd: int, path: int, path_len: int) -> HostResult;
+	def clock_res_get(clock_id: int, resolution: int) -> HostResult;
+	def clock_time_get(clock_id: int, precision: long, time: int) -> HostResult;
+	def sched_yield() -> int;
+	def proc_exit(code: int) -> HostResult {
+		// TODO: Is this how we want to handle this case?
```

Comment:
Not sure what this comment means.

Reply:
This was just me keeping a reminder that I wanted to make sure we're okay with returning a `HostResult` directly.